customers/schema/queries.py

```diff
@@ -179,6 +208,17 @@ def resolve_berth_profiles(self, info, **kwargs):
 
         qs = CustomerProfile.objects
 
+        # Check if Helsinki Profiles filters are used in the query, if yes, query the
+        # profile ids first from there
+        if not set(kwargs.keys()).isdisjoint(set(HELSINKI_PROFILES_FILTERS)):
+            profile_token = kwargs.pop("api_token", None)
+            if not profile_token:
+                raise VenepaikkaGraphQLError(
+                    "Cannot filter by Helsinki Profile fields without API Token"
+                )
+            ids = _get_ids_from_profile_service(kwargs, profile_token)
+            qs = qs.filter(id__in=ids)
```

Comment:
Would we need to have some extra logic here in case the `sort_by` parameter was used? I'm not sure if using `filter` is enough since by default `CustomerProfile` is ordered by `id`. I think we need to sort the `qs` by sorting them to the same order as the list of `ids` we get from Helsinki profile.

Reply:
https://stackoverflow.com/questions/27725703/rearranging-list-based-on-order-of-another-list

Reply:
In this implementation the `ids = _get_ids_from_profile_service(kwargs, profile_token)` is a single page list of all the ids. I think it is also needed to be so, because otherwise we would not be able to get the total count etc. right? Would it be too much to sort them something like is introduced here: https://stackoverflow.com/questions/27725703/rearranging-list-based-on-order-of-another-list - a basic sort list with another list approach.